sql-db/narayana-transactions/src/test/java/io/quarkus/ts/transactions/OracleTransactionGeneralUsageIT.java

```diff
@@ -13,17 +13,32 @@
 @DisabledIfSystemProperty(named = "ts.arm.missing.services.excludes", matches = "true", disabledReason = "https://github.com/quarkus-qe/quarkus-test-suite/issues/2022")
 public class OracleTransactionGeneralUsageIT extends TransactionCommons {
 
-    static final int ORACLE_PORT = 1521;
+    private static final int ORACLE_PORT = 1521;
+    private static final String DATABASE = "mydb";
+    private static final String SEPARATOR = ",";
+    /**
+     * Create database 'mydb' and 'mydb2' because Oracle prevents 2PC when the same XA resource is detected.
+     * When a one transaction is executed on 2 different databases, it is clear that 2PC is required.
+     */
+    private static final String DATABASES = DATABASE + SEPARATOR + DATABASE + 2;
 
     @Container(image = "${oracle.image}", port = ORACLE_PORT, expectedLog = "DATABASE IS READY TO USE!")
-    static OracleService database = new OracleService();
+    static OracleService database = new OracleService().withDatabase(DATABASES);
 
     @QuarkusApplication
     static RestService app = new RestService().withProperties("oracle.properties")
             .withProperty("quarkus.otel.exporter.otlp.traces.endpoint", jaeger::getCollectorUrl)
             .withProperty("quarkus.datasource.username", database.getUser())
             .withProperty("quarkus.datasource.password", database.getPassword())
-            .withProperty("quarkus.datasource.jdbc.url", database::getJdbcUrl);
+            .withProperty("quarkus.datasource.jdbc.url", () -> {
+                // now that both databases are created, get the JDBC URL for the first database
+                // TODO: add 'getJdbcUrl(String databaseName)' method to the framework
```

Comment:
It looks like this PR is not urgent. Maybe we should first make the changes in the FW?

Reply:
> It looks like this PR is not urgent. Maybe we should first make the changes in the FW?

ok